WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SiteCreationService.java

```diff
@@ -0,0 +1,338 @@
+            String siteTagline,
+            String siteSlug,
+            String siteTheme) {
+        Intent intent = new Intent(context, SiteCreationService.class);
+        intent.putExtra(ARG_SITE_TITLE, siteTitle);
+        intent.putExtra(ARG_SITE_TAGLINE, siteTagline);
+        intent.putExtra(ARG_SITE_SLUG, siteSlug);
+        intent.putExtra(ARG_SITE_THEME, siteTheme);
+        context.startService(intent);
+    }
+
+    public SiteCreationService() {
+        super(OnSiteCreationStateUpdated.class);
+    }
+
+    @Override
+    protected OnSiteCreationStateUpdated getCurrentStateEvent() {
+        return new OnSiteCreationStateUpdated(mSiteCreationPhase);
+    }
+
+    @Override
+    public boolean isInProgress() {
+        return mSiteCreationPhase != SiteCreationPhase.IDLE
+                && mSiteCreationPhase != SiteCreationPhase.SUCCESS
+                && mSiteCreationPhase != SiteCreationPhase.FAILURE;
+    }
+
+    @Override
+    public boolean isError() {
+        return mSiteCreationPhase == SiteCreationPhase.FAILURE;
+    }
+
+    @Override
+    public Notification getNotification() {
+        switch (mSiteCreationPhase) {
+            case NEW_SITE:
+                return getProgressNotification(25, "Site creation in: " + mSiteCreationPhase.name());
+            case FETCHING_NEW_SITE:
+                return getProgressNotification(50, "Site creation in: " + mSiteCreationPhase.name());
+            case SET_TAGLINE:
+                return getProgressNotification(75, "Site creation in: " + mSiteCreationPhase.name());
+            case SET_THEME:
+                return getProgressNotification(100, "Site creation in: " + mSiteCreationPhase.name());
+            case SUCCESS:
+                return getSuccessNotification("Site created!");
+            case FAILURE:
+                return getFailureNotification("Site creation failed :(");
+        }
+
+        return null;
+    }
+
+    private void setState(SiteCreationPhase siteCreationPhase) {
+        mSiteCreationPhase = siteCreationPhase;
+        notifyState();
+
+        if (siteCreationPhase == SiteCreationPhase.FAILURE || siteCreationPhase == SiteCreationPhase.SUCCESS) {
+            stopSelf();
+        }
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        ((WordPress) getApplication()).component().inject(this);
+
+        AppLog.i(T.MAIN, "SiteCreationService > Created");
+        mDispatcher.register(this);
+
+        // TODO: Recover any site creations that were interrupted by the service being stopped?
+    }
+
+    @Override
+    public void onDestroy() {
+        mDispatcher.unregister(this);
+        AppLog.i(T.MAIN, "SiteCreationService > Destroyed");
+        super.onDestroy();
+    }
+
+    private Intent getPendingIntent() {
+        Intent intent = new Intent(this, NewBlogActivity.class);
+//        intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
+//        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
```

Comment:
I guess this is only used for the Progress notification (if the user taps on it, they will be taken to the `NewBlogActivity`), and as such we don't need to create a new task stack, etc. If that's its only purpose I think we can safely remove these commented lines.

Reply:
I haven't fully implemented the notification tapping behavior and that's why those are still in comments. I'll remove the comments and we can fine tune the notification tapping behavior in a subsequent PR, is that OK?

Reply:
cool!